src/tests/unit/sync/syncimages_ring.f90

```diff
@@ -0,0 +1,25 @@
+! SYNC IMAGES([this_image - 1, this_image + 1]) with the STAT=STAT_STOPPED_IMAGE
+! specifier and wrap around of image numbers.  The test is intended to check
+! that syncing in a ring with a stopped image still terminates all images.
+
+program sync_images_ring
+  use, intrinsic:: iso_fortran_env
+  implicit none
+
+  integer :: stat_var = 0
+
+  if (num_images() .lt. 2) error stop "Need at least two images to test."
+
+  associate (me => this_image())
+    if (me /= 1) then
```

Comment:
Just to make it more clear that image 1 is expected to reach termination before the other images since it is not part of the `sync images(*,...)` call. It might be a dumb/infeasable idea, I'll test it and see what happens. Otherwise I'll just add some additional comments to clarify.